### PR TITLE
Sync bilibili favlist

### PR DESCRIPTION
**Pull request by zzztttkkk**

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -272,17 +268,26 @@ func New() *cli.App {
 }
 
 func download(c *cli.Context, videoURL string) error {
+	errcontinue := c.Bool("errorcontinue")
+
+	cookie, err := utils.ReadCookie(c.String("cookie"))
+	if err != nil {
+		return nil
+	}
```
Comment (iawia002) [inline app/app.go:276]: We can make the `cookie` as a parameter, so there's no need to parse it repeatedly.

File: downloader/downloader.go [hunk 2]
```diff
@@ -694,8 +694,18 @@ func (downloader *Downloader) Download(data *extractors.Data) error {
...
+
 	if stream.Ext != "mp4" || stream.NeedMux {
-		return utils.MergeFilesWithSameExtension(parts, mergedFilePath)
+		err = utils.MergeFilesWithSameExtension(parts, mergedFilePath)
+	} else {
+		err = utils.MergeToMP4(parts, mergedFilePath, title)
+	}
+	if err != nil {
+		return nil
+	}
```
Comment (iawia002) [inline downloader/downloader.go:705]: Why do we ignore this error directly?

File: extractors/bilibili/bilibili_test.go [hunk 3]
```diff
@@ -95,3 +97,7 @@ func TestBilibili(t *testing.T) {
 		})
 	}
 }
+
+func TestPath(t *testing.T) {
+	fmt.Println(filepath.Base(`vids\自制捕蚊子陷阱.mp4`))
+}
```
Comment (iawia002) [inline extractors/bilibili/bilibili_test.go:103]:
```suggestion
```

File: downloader/downloader.go (hunk 2 above)
Comment (zzztttkkk) [reply downloader/downloader.go:705]: ...this is a mistake.

**Pull request closed without merging**